src/data_source/fs.rs

```diff
@@ -158,10 +158,19 @@ where
     /// If there is already data at `path`, it will be archived.
     ///
     /// The [FileSystemDataSource] will manage its own persistence synchronization.
-    pub async fn create(path: &Path, provider: P) -> anyhow::Result<Self> {
-        FetchingDataSource::builder(FileSystemStorage::create(path).await?, provider)
-            .build()
-            .await
+    pub async fn create(
+        path: &Path,
+        provider: P,
+        fetch_rate_limit: Option<usize>,
```

Comment:
Remove this argument, if you want to configure a rate limit you should just create this from a builder

Reply:
we call the create function of FetchingDataSource in sequencer which creates a  default builder

Reply:
Then we should change that to use a builder